core/src/main/java/org/apache/iceberg/BaseRewriteFiles.java

```diff
@@ -44,6 +45,11 @@ protected String operation() {
     return DataOperations.REPLACE;
   }
 
+  @Override
+  protected Long sequenceNumber() {
+    return replaceSequenceNumber;
```

Comment:
With this approach, I think we need a validation that none of the data or delete files that are being replaced have sequence numbers newer than the override sequence number.